### PR TITLE
[Inference] Fix bug in ChatGLM2 Tensor Parallelism

### PR DESCRIPTION
**Pull request by CjhHa1**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.
Fix bug in chatglm2 tp inference

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/inference/tensor_parallel/modeling/chatglm2.py [hunk 1]
```diff
@@ -466,19 +465,18 @@ def chatglm_flash_attn_kvcache_forward(
...
             copy_kv_to_mem_cache(
                 infer_state.decode_layer_id,
                 key_layer,
                 value_layer,
                 infer_state.context_mem_index,
                 infer_state.cache_manager,
             )
-
-            attn_output = torch.empty_like(query_layer.view(-1, self.projection_size))
+            attn_output = torch.empty_like(query_layer.contiguous().view(-1, self.projection_size))
```
Comment (tiandiao123) [inline colossalai/inference/tensor_parallel/modeling/chatglm2.py:479]: why is that ? why we need it to be contiguous?

Comment (CjhHa1) [reply colossalai/inference/tensor_parallel/modeling/chatglm2.py:479]: encountered Runtime error when not using `contiguous()`

File: tests/test_infer/test_chatglm2_infer.py [hunk 2]
```diff
@@ -13,13 +13,12 @@
 from colossalai.testing import clear_cache_before_run, parameterize, rerun_if_address_is_in_use, spawn
 
 try:
-    import lightllm
```
Comment (ver217) [inline tests/test_infer/test_chatglm2_infer.py:16]: If removing this, `HAS_LIGHTLLM_KERNEL` may be always true

Comment (CjhHa1) [reply tests/test_infer/test_chatglm2_infer.py:16]: Fixed

**Review by tiandiao123 — APPROVED** [review]

**Review by ver217 — APPROVED** [review]

**Pull request merged**